db-engine/src/main/java/com/splicemachine/db/impl/sql/catalog/DataDictionaryImpl.java

```diff
@@ -9321,6 +9333,32 @@ else if( routinePermsKey.getGrantee().equals( sd.getAuthorizationId()))
         else if (key instanceof SchemaPermsDescriptor) {
             SchemaPermsDescriptor schemaPermsKey = (SchemaPermsDescriptor) key;
             permissions = getUncachedSchemaPermsDescriptor(schemaPermsKey);
+
+            if( permissions == null)
+            {
+                try {
+                    // The owner has all privileges unless they have been revoked.
+                    SchemaDescriptor sd = getSchemaDescriptor(schemaPermsKey.getSchemaUUID(), ConnectionUtil.getCurrentLCC().getTransactionExecute());
+
+                    if (metadataAccessRestrictionEnabled && sd.isSystemViewSchema() ||
+                            !metadataAccessRestrictionEnabled && sd.isSystemSchema()) {
+                        // allow read access to public
+                        permissions = new SchemaPermsDescriptor(this,
+                                schemaPermsKey.getGrantee(),
+                                (String) null,
```

Comment:
Could we replace this `null` with a more meaningful constant?

Reply:
This is the current way it is implemented and widely used, I prefer not to change in this PR.